homeassistant/components/config/config_entries.py

```diff
@@ -388,6 +412,7 @@ def entry_json(entry: config_entries.ConfigEntry) -> dict:
         "source": entry.source,
         "state": entry.state.value,
         "supports_options": supports_options,
+        "supports_remove_device": entry.supports_remove_device,
```

Comment:
We may not want to include this in all responses, we should maybe do an `entry_json_extended` with all flags.

Reply:
Why not always include it?

Reply:
@bramkragten suggested it adds unwanted overhead to keep adding flags to the responses, when they're maybe only used in some special cases.

Reply:
Device Registry is always kept in memory when HA is running. We should probably propose a slim version and use that in many places in the UI.